scheduler/scheduler-server/src/test/java/functionaltests/recover/TaskReconnectionToDownNodeTest.java

```diff
@@ -0,0 +1,247 @@
+import org.ow2.proactive.scheduler.common.exception.PermissionException;
+import org.ow2.proactive.scheduler.common.exception.UnknownJobException;
+import org.ow2.proactive.scheduler.common.job.JobId;
+import org.ow2.proactive.scheduler.common.job.JobResult;
+import org.ow2.proactive.scheduler.common.job.JobState;
+import org.ow2.proactive.scheduler.common.task.TaskInfo;
+import org.ow2.proactive.scheduler.common.task.TaskResult;
+import org.ow2.proactive.scheduler.common.task.TaskState;
+import org.ow2.proactive.scheduler.common.task.TaskStatus;
+import org.ow2.proactive.scheduler.core.properties.PASchedulerProperties;
+
+import functionaltests.nodesrecovery.RecoverInfrastructureTestHelper;
+import functionaltests.utils.SchedulerFunctionalTestWithCustomConfigAndRestart;
+import functionaltests.utils.SchedulerTHelper;
+import functionaltests.utils.TestNode;
+import functionaltests.utils.TestScheduler;
+
+
+/**
+ * This test verifies that when
+ * @author ActiveEon Team
+ * @since 20/09/17
+ */
+public class TaskReconnectionToDownNodeTest extends SchedulerFunctionalTestWithCustomConfigAndRestart {
+
+    private static final URL JOB_DESCRIPTOR = TaskReconnectionToDownNodeTest.class.getResource("/functionaltests/descriptors/Job_TaskReconnectionOnRestart_25_Parallel_Tasks.xml");
+
+    private static final URL RM_CONFIGURATION_START = TaskReconnectionToDownNodeTest.class.getResource("/functionaltests/config/functionalTRMProperties-clean-db.ini");
+
+    private static final URL RM_CONFIGURATION_RESTART = TaskReconnectionToDownNodeTest.class.getResource("/functionaltests/config/functionalTRMProperties-keep-db.ini");
+
+    private static final int NUMBER_OF_NODES = 25;
+
+    private static final int NUMBER_OF_REPLICATE_TASKS = 25;
+
+    private static final int RESTART_SCHEDULER_INTER_TIME_IN_MILLISECONDS = 1000;
+
+    private static final String LAST_REPLICATE_TASK_NAME = "Groovy_Task26";
+
+    private List<TestNode> nodes;
+
+    private Map<Long, String> taskExecutionHostnamePerTaskId;
+
+    private static final URL SCHEDULER_CONFIGURATION_START = TaskReconnectionWithForkedTaskExecutorTest.class.getResource("/functionaltests/config/functionalTSchedulerProperties.ini");
+
+    private static final URL SCHEDULER_CONFIGURATION_RESTART = TaskReconnectionWithForkedTaskExecutorTest.class.getResource("/functionaltests/config/functionalTSchedulerProperties-updateDB.ini");
+
+    protected URL getSchedulerStartConfigurationURL() {
+        return SCHEDULER_CONFIGURATION_START;
+    }
+
+    protected URL getSchedulerReStartConfigurationURL() {
+        return SCHEDULER_CONFIGURATION_RESTART;
+    }
+
+    @Before
+    public void startDedicatedScheduler() throws Exception {
+        RMFactory.setOsJavaProperty();
+        schedulerHelper = new SchedulerTHelper(false,
+                                               new File(getSchedulerStartConfigurationURL().toURI()).getAbsolutePath(),
```

Comment:
I believe you can do:
getSchedulerStartConfigurationURL().getPath()

Reply:
I can do getSchedulerStartConfigurationURL().getPath()).getAbsolutePath(). Is it better?

Reply:
I am fine with anything..
I just meant that "getSchedulerStartConfigurationURL().getPath()" could be enought as an argument
...without new File(). but i am not sure so ignore in case